src/renderers/shared/fiber/ReactFiberTreeReflection.js

```diff
@@ -76,6 +76,12 @@ exports.findCurrentHostFiber = function(component : ReactComponent<any, any, any
 
   let didTryOtherTree = false;
 
+  // If the component doesn't have a child we first check the alternate to see
+  // if it has any and if so, if those were just recently inserted.
+  if (!parent.child && parent.alternate) {
```

Comment:
We agreed at lunch that this check is wrong and that we can't currently distinguish when a new child is inserted at the start of a fragment with this strategy.